src/context/ModalProvider/types.ts

```diff
@@ -41,18 +41,21 @@ export type ModalActions<T extends keyof Modals> = OpenModalType<T> | CloseModal
 
 export type OpenModalType<T extends keyof Modals> = {
   type: typeof OPEN_MODAL
+  key: T
   props: ModalProps<T>
 }
 
 export type CloseModalType = {
   type: typeof CLOSE_MODAL
+  key: keyof Modals
 }
-
 export type ModalProps<T extends keyof Modals> = React.ComponentProps<Modals[T]>
 
-export type ModalState<T extends keyof Modals> = {
-  props?: ModalProps<T>
-  isOpen: boolean
+export type ModalState = {
+  [K in keyof Modals]: {
```

Comment:
The guts of this PR - modals are now indexed per modal in a *global* reducer vs. many reducers (which were useless, as only one reducer was effective)